src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs

```diff
@@ -0,0 +1,662 @@
+            &mut self.wai_consumer,
+            Self::WAI_VALVE_TRANSFER_SPEED,
+        );
+
+        // Check if HIGH or LOW pressure
+        if self.wai_consumer_pressure()
+            <= Pressure::new::<bar>(Self::WAI_MIN_PRESSURE) + context.ambient_pressure()
+            && !self.is_wai_valve_closed()
+        {
+            self.wai_high_pressure = false;
+            self.wai_low_pressure = true;
+            self.wai_has_fault = true;
+        } else if self.wai_consumer_pressure()
+            >= Pressure::new::<bar>(Self::WAI_MAX_PRESSURE) + context.ambient_pressure()
+            && !self.is_wai_valve_closed()
+        {
+            self.wai_high_pressure = true;
+            self.wai_low_pressure = false;
+            // High pressure doesn't turn the FAULT light ON
+        } else {
+            self.wai_high_pressure = false;
+            self.wai_low_pressure = false;
+        }
+    }
+}
+
+// FWC FAILURES TO IMPLEMENT
+// WING A.ICE SYS FAULT
+// WING A.ICE L(R) VALVE OPEN
+// WING A.ICE OPEN ON GND
+// WING A.ICE L(R) HI PR
+
+// The complex includes both WingAntiIceSystem parts.
+// Each WingAntiIceSystem contains a consumer, a valve,
+// an exhaust and a valve controller.
+// There is one (shared) ground sense relay.
+pub struct WingAntiIceComplex {
+    // Left and Right wing
+    wai_systems: [WingAntiIceSystem; 2],
+    wai_relay: WingAntiIceRelay,
+    wai_system_has_fault: bool,
+    wai_system_on: bool,
+    wai_selected: bool,
+
+    wai_on_id: VariableIdentifier,
+    wai_selected_id: VariableIdentifier,
+    wai_fault_id: VariableIdentifier,
+    wai_ground_timer_id: VariableIdentifier,
+    wai_left_pressure_id: VariableIdentifier,
+    wai_right_pressure_id: VariableIdentifier,
+    wai_left_temperature_id: VariableIdentifier,
+    wai_right_temperature_id: VariableIdentifier,
+    wai_left_valve_closed_id: VariableIdentifier,
+    wai_right_valve_closed_id: VariableIdentifier,
+
+    wai_left_high_pressure_id: VariableIdentifier,
+    wai_right_high_pressure_id: VariableIdentifier,
+    wai_left_low_pressure_id: VariableIdentifier,
+    wai_right_low_pressure_id: VariableIdentifier,
+}
+impl WingAntiIceComplex {
+    pub fn new(context: &mut InitContext) -> Self {
+        Self {
+            wai_systems: [WingAntiIceSystem::new(), WingAntiIceSystem::new()],
+            wai_relay: WingAntiIceRelay::new(),
+            wai_system_has_fault: false,
+            wai_system_on: false,
+            wai_selected: false,
+
+            wai_on_id: context.get_identifier("PNEU_WING_ANTI_ICE_SYSTEM_ON".to_owned()),
+            wai_selected_id: context
+                .get_identifier("PNEU_WING_ANTI_ICE_SYSTEM_SELECTED".to_owned()),
+            wai_fault_id: context.get_identifier("PNEU_WING_ANTI_ICE_HAS_FAULT".to_owned()),
+            wai_ground_timer_id: context
+                .get_identifier("PNEU_WING_ANTI_ICE_GROUND_TIMER".to_owned()),
+
+            wai_left_valve_closed_id: context
```

Comment:
Move the writing of these simvars into `WingAntiIceSystem` and adjust the simvars according to my previous comment.
You could pass the id of the system when calling `WingAntiIceSystem::new()` to create the appropiate simvars:
```suggestion
            wai_systems: [WingAntiIceSystem::new(1), WingAntiIceSystem::new(2)],
```
or
```suggestion
            wai_systems: [WingAntiIceSystem::new('L'), WingAntiIceSystem::new('R')],
```

Reply:
I have done the same for the stuff written by WingAntiIceRelay